packages/react-storage/src/components/StorageBrowser/views/LocationDetailView/useLocationDetailView.ts

```diff
@@ -102,20 +120,30 @@ export function useLocationDetailView(
   const onRefresh = () => {
     if (hasInvalidPrefix) return;
     handleReset();
-    handleList({ prefix, options: { ...listOptions, refresh: true } });
+    handleList({ config, prefix, options: { ...listOptions, refresh: true } });
     dispatchStoreAction({ type: 'RESET_LOCATION_ITEMS' });
   };
 
+  const onSearch = (query: string, includeSubfolders: boolean) => {
+    if (hasInvalidPrefix) return;
+    const searchOptions = {
+      ...listOptions,
+      delimiter: includeSubfolders ? undefined : listOptions.delimiter,
+      search: { query, filterKey: 'key' as const },
```

Comment:
Will `key` always be the `filterKey`? If so do we need to pass it an option at all?

Reply:
It could be different for each view. for example: List Locations uses the filterKey `prefix`. I think we might actually want to make this a list to support multi column search, but we don't have that req currently.